From e04281d50fb9b487a9330e1d6f666457874f7e52 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Thu, 6 Aug 2020 19:24:51 +0700
Subject: [PATCH 427/441] ARM: dts: r8a77470: Fix issue cannot detect USB2.0
 devices

This patch corrects ehci1, ohci1 and usbphy settings to fix USB 2.0
issue on RZ/G1C platform (R8A77470).

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 54 +++++++++++++++++++++++------------------
 1 file changed, 30 insertions(+), 24 deletions(-)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 70141e2..d848f96 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -395,28 +395,29 @@
 		hsusb0: hsusb@e6590000 {
 			compatible = "renesas,usbhs-r8a77470",
 				     "renesas,rcar-gen2-usbhs";
-			reg = <0 0xe6590000 0 0x100>;
-			interrupts = <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&mstp7_clks R8A77470_CLK_USBHS0>;
+			reg = <0 0xe6590000 0 0x0ff>;
+			interrupts = <0 107 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp7_clks R8A77470_CLK_USB_EHCI_0>;
 			dmas = <&usb_dmac00 0>, <&usb_dmac00 1>,
 			       <&usb_dmac10 0>, <&usb_dmac10 1>;
 			dma-names = "ch0", "ch1", "ch2", "ch3";
 			renesas,buswait = <4>;
-			phys = <&usb0 1>;
+			phys = <&usb0 0>;
 			phy-names = "usb";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
 
 		usbphy0: usb-phy@e6590100 {
 			compatible = "renesas,usb-phy-r8a77470",
 				     "renesas,rcar-gen2-usb-phy";
-			reg = <0 0xe6590100 0 0x100>;
+			reg = <0 0xe6590100 0 0x0ff>,
+				<0 0xee080200 0 0x6ff>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			clocks = <&mstp7_clks R8A77470_CLK_USBHS0>;
-			clock-names = "usbhs";
-			power-domains = <&cpg_clocks>;
+			clock-names = "hsusb0";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 
 			usb0: usb-channel@0 {
@@ -445,18 +446,24 @@
 		usbphy1: usb-phy@e6598100 {
 			compatible = "renesas,usb-phy-r8a77470",
 				     "renesas,rcar-gen2-usb-phy";
-			reg = <0 0xe6598100 0 0x100>;
+			reg = <0 0xe6598100 0 0x0ff>,
+				<0 0xee0c0200 0 0x6ff>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			clocks = <&mstp7_clks R8A77470_CLK_USBHS1>;
-			clock-names = "usbhs";
-			power-domains = <&cpg_clocks>;
+			clock-names = "hsusb1";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 
 			usb1: usb-channel@0 {
 				reg = <0>;
 				#phy-cells = <1>;
 			};
+
+			usb2: usb-channel@2 {
+				reg = <2>;
+				#phy-cells = <1>;
+			};
 		};
 
 		usb_dmac00: dma-controller@e65a0000 {
@@ -809,27 +816,26 @@
 			status = "disabled";
 		};
 
-		ohci1: usb@ee0c0000 {
+		ohci1: ohci-platform.1@ee0c0000 {
 			compatible = "generic-ohci";
-			reg = <0 0xee0c0000 0 0x100>;
-			interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>;
+			reg = <0 0xee0c0000 0 0x0ff>;
+			interrupts = <0 113 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_USB_EHCI_1>;
-			phys = <&usb0 1>, <&usb2_phy1>, <&usb1 0>;
+			phys = <&usb2 0>;
 			phy-names = "usb";
-			power-domains = <&cpg_clocks>;
-			status = "disabled";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disable";
 		};
 
-		ehci1: usb@ee0c0100 {
+		ehci1: ehci-platform.1@ee0c0100 {
 			compatible = "generic-ehci";
-			reg = <0 0xee0c0100 0 0x100>;
-			interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>;
+			reg = <0 0xee0c0100 0 0x0ff>;
+			interrupts = <0 113 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_USB_EHCI_1>;
-			phys = <&usb0 1>, <&usb2_phy1>, <&usb1 0>;
+			phys = <&usb2 0>;
 			phy-names = "usb";
-			companion = <&ohci1>;
-			power-domains = <&cpg_clocks>;
-			status = "disabled";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disable";
 		};
 
 		usb2_phy1: usb-phy@ee0c0200 {
-- 
2.7.4

